[Chang Chuan Hao] iP

Chang-CH opened the pull request:
# Duk project

> "It works" - me ([source](https://github.com/nus-cs2103-AY2122S1/ip/pull/19))

This project features GitHub Flavoured Mark Down, such as

- bullet
- points

1. numbered
2. lists

```java
public class FencedCodeBlock {
    public static void main(String[] args) {
      System.out.println("With syntax highlighting");
    }
}
```

- [ ] task
- [ ] lists
- [x] checked

Em🙃ji

> Block
> Quote

Hyper[link](https://github.com/nus-cs2103-AY2122S1/ip/pull/19)

inline `code`

~~text~~ **formatting** w/ *italic*

Commit:
Add Gradle support

Chang-CH committed:
Added Greet, Echo, Exit

Chang-CH committed:
Update JDK version

Chang-CH committed:
Reverted to switch statements

Chang-CH committed:
Completed Add, List

Chang-CH committed:
Completed Mark as Done

Chang-CH committed:
Update Duke.java

Chang-CH committed:
Completed ToDos, Events, Deadlines

Chang-CH committed:
Added Text UI testing

Chang-CH committed:
Fix Todo entries

Chang-CH committed:
Completed Handle Errors

Chang-CH committed:
Completed Delete

Chang-CH committed:
Fix trailing space in tests

Chang-CH committed:
Merge branch 'add-gradle-support'

Chang-CH committed:
Completed Level 7. Save

Chang-CH committed:
Completed Level 8. Dates and Times

Chang-CH committed:
Merge branch 'branch-Level-7'

Chang-CH committed:
Merge branch 'branch-Level-8'

Chang-CH committed:
Fixed event date processing

Chang-CH committed:
Fix merge errors

Chang-CH committed:
Revert "Fix merge errors"

This reverts commit d6bb4895a601f544d2e288d440129a956e4e1847.

Chang-CH committed:
Fix date commit conflict with save regex

Chang-CH committed:
Completed Use More OOP

Chang-CH committed:
Completed Organize into Packages

Chang-CH committed:
Complete Add JUnit Tests

Chang-CH committed:
Update build scripts

Chang-CH committed:
Add javadoc comments

Chang-CH committed:
Add javadocs

Chang-CH committed:
Updated coding standard

Chang-CH committed:
Completed Level 9 Find.

Chang-CH committed:
Revert merge branch 'branch-A-CodingStandard'

Chang-CH committed:
Merge branch 'branch-A-CodingStandard'

Chang-CH committed:
Merge branch 'branch-Level-9'

Chang-CH committed:
Merge branch 'branch-A-JavaDoc'

Chang-CH committed:
Complete A-Gradle

Chang-CH committed:
Fix checkstyle audit

Chang-CH committed:
Complete Level-10

Chang-CH committed:
Add padding

yatbun commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,78 @@
+
+import java.time.LocalDate;
+
+// Duke Commands
```
I like how you have a structured way of importing classes 👍

yatbun commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,78 @@
+        case "find":
+            return new FindCommand(input.substring(5));
+        case "event":
+            params = input.split("/at");
```
Perhaps consider using `.split(" /at/ ")` to reduce the confusing subsequent use of `subString()`.

yatbun commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,145 @@
+    protected void showLoadingError(String error) {
+        System.out.println("OOPS!!! I'm sorry, but I don't know what that means :-(");
+        if (dialogContainer != null) {
+            printDialog("OOPS!!! I'm sorry, but I don't know what that means :-(");
```
Common error messages could perhaps be extracted out to prevent duplicate code.

yatbun commented on src/main/java/duke/command/FindCommand.java:
```diff
@@ -0,0 +1,48 @@
+     */
+    public FindCommand(String regex) {
+        this.regex = regex;
+    }
```
Would suggest putting a line break between the end of this method and the javaDoc for the next to improve readability.

yatbun reviewed (commented):
Overall, your code looks amazing 👍 you followed the coding standard very closely. Some cosmetic changes were requested.

Chang-CH committed:
update gradle

Chang-CH committed:
Complete A-Assertions

Assertions missing for all classes and methods

Assertions help verify assumptions about the program at runtime.

Let's add assertions to verify arguments of private/protected methods

There are several private/protected methods that should only recieve certain inputs.

Refer to the cs2103T project requirements.

Chang-CH committed:
Complete A-Assertions

Assertions missing for all classes and methods

Assertions help verify assumptions about the program at runtime.

Let's add assertions to verify arguments of private/protected methods

There are several private/protected methods that should only recieve certain inputs.

Refer to the project requirements [here](https://nus-cs2103-ay2122s1.github.io/website/schedule/week5/project.html#a-assertions)

Chang-CH committed:
Complete A-Streams

Streams could be used to satisfy project increments.

Streams make code simpler and easy to read.

Let's process GUI images as streams.

Stream usage is preferable.

Chang-CH committed:
Complete A-CodeQuality

Chang-CH committed:
Complete A-CodeQuality

Various files: various methods are not in line with code quality guides

Matching code quality guidelines make the project more readable to incoming developers

Update code quality, magic numbers changed to named constants, unused methods removed.

To keep in line with code quality proposed.

Chang-CH committed:
Revert "Complete A-CodeQuality"

This reverts commit 8c529f2406db851ad8918a146418b0dfbf7d7bb0.

Chang-CH committed:
Merge pull request #1 from Chang-CH/branch-A-CodeQuality

Complete A-CodeQuality

Chang-CH committed:
Merge pull request #2 from Chang-CH/branch-A-Streams

Complete A-Streams

Chang-CH committed:
Merge branch 'branch-A-Assertions'

Chang-CH committed:
Complete C-Help

Chang-CH committed:
Merge branch 'branch-A-Assertions' of https://github.com/Chang-CH/ip into branch-A-Assertions

Chang-CH committed:
Merge branch 'branch-A-Assertions'

Chang-CH committed:
Add additional assertions

Chang-CH committed:
Merge branch 'master' into branch-A-Assertions

Chang-CH committed:
Merge pull request #4 from Chang-CH/branch-A-Assertions

Merge branch A-Assertions

Chang-CH committed:
Fix Parser imports

Chang-CH committed:
Complete A-CI

Chang-CH committed:
Fix coding standard

Chang-CH committed:
Update README.md

Chang-CH committed:
Update Ui.java

Chang-CH committed:
Update Ui.java

Removed unused imports

Chang-CH committed:
Merge pull request #5 from Chang-CH/branch-A-Streams

Update Ui.java

Chang-CH committed:
Improve code quality

Chang-CH committed:
Merge branch 'master' into branch-A-CodeQuality

Chang-CH committed:
Update Storage.java

Chang-CH committed:
Merge pull request #6 from Chang-CH/branch-A-CodeQuality

Improve code quality

Chang-CH committed:
Update visuals

Chang-CH committed:
Add user guide

Chang-CH committed:
Add folder creation support

Chang-CH committed:
Set theme jekyll-theme-midnight

Chang-CH committed:
Update coding standard

Chang-CH committed:
Merge branch 'master' of https://github.com/Chang-CH/ip

Commit:
fix Mac OS path issue

Commit:
Improve exception handling

Commit:
Fix peer evaluation issues
